src/groups/mqb/mqbc/mqbc_clusternodesession.cpp

```diff
@@ -157,13 +144,19 @@ bool ClusterNodeSession::removePartitionSafe(int partitionId)
     // PRECONDITIONS
     BSLS_ASSERT_SAFE(dispatcher()->inDispatcherThread(this));
 
-    if (d_primaryPartitions.end() == bsl::find(d_primaryPartitions.begin(),
-                                               d_primaryPartitions.end(),
-                                               partitionId)) {
+    bsl::vector<int>::iterator it = bsl::find(d_primaryPartitions.begin(),
+                                              d_primaryPartitions.end(),
+                                              partitionId);
+    if (it == d_primaryPartitions.end()) {
         return false;  // RETURN
     }
 
-    removePartitionRaw(partitionId);
+    d_primaryPartitions.erase(it);
```

Comment:
Sanity check:

`removePartitionRaw` looks like it removed `partitionId` and everything following it.  That's strange.  This change only removes `partitionId`.

Was this change intentional?